Assets/UGSSamples/FriendsSample/Scripts/Common/Managers/RelationshipsManager.cs

```diff
@@ -216,7 +225,7 @@ async Task<bool> SendFriendRequest(string playerName)
                 //We add the friend by name in this sample but you can also add a friend by ID using AddFriendAsync
                 var relationship = await FriendsService.Instance.AddFriendByNameAsync(playerName);
                 Debug.Log($"Friend request sent to {playerName}.");
-                return relationship.Type == RelationshipType.FRIEND_REQUEST;
+                return relationship.Type == RelationshipType.FRIEND_REQUEST || relationship.Type == RelationshipType.FRIEND;

```

Comment:
Why would we sent a request to someone that is already a friend?
If we want to cover this use case, should we handle it better, and have a message "Already in your friend list" and deny to send the friend request?

Reply:
Agreed. There may be some weirdness with this change when adding users who are already Friends. 

My changes were addressing this flow: 

Step 1) Player A sends a Friend Request to player B (via add friend popup)
Step 2) Player B sends a Friend Request to Player A (via add friend popup)

The error happens on Player B client: In this scenario, the relationship.Type returns FRIEND and not FRIEND_REQUEST so the error flow is shown on the popup but the friend request goes through for Player A.  Player B's UI is broken until a reset (maybe some other flow can correct it)